pkg/kubelet/kubelet_pods.go

```diff
@@ -1503,13 +1503,24 @@ func (kl *Kubelet) generateAPIPodStatus(pod *v1.Pod, podStatus *kubecontainer.Po
 	// ensure the probe managers have up to date status for containers
 	kl.probeManager.UpdatePodStatus(pod.UID, s)
 
-	// preserve all conditions not owned by the kubelet
+	// preserve all conditions not owned nor shared by the kubelet
 	s.Conditions = make([]v1.PodCondition, 0, len(pod.Status.Conditions)+1)
 	for _, c := range pod.Status.Conditions {
-		if !kubetypes.PodConditionByKubelet(c.Type) {
+		if !kubetypes.PodConditionByKubelet(c.Type) && !kubetypes.PodConditionSharedByKubelet(c.Type) {
 			s.Conditions = append(s.Conditions, c)
 		}
 	}
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
+		// copy over the pod disruption conditions from state which is already
+		// updated during the eviciton (due to either node resource pressure or
+		// node graceful shutdown). We do not re-generate the conditions based
+		// on the container statuses as they are added based on one-time events.
+		if _, condition := podutil.GetPodConditionFromList(oldPodStatus.Conditions, v1.AlphaNoCompatGuaranteeDisruptionTarget); condition != nil {
+			s.Conditions = append(s.Conditions, *condition)
```

Comment:
why are we excluding the condition above, then appending the condition from oldPodStatus here?

Reply:
no particular reason, works fine if the same pattern is applied and it gets simpler. Pushed changes